### ISSUE TITLE
SOAP Scanner operations are overwritten in sites tree.

### ISSUE THREAD

**Issue by arcocap4z** [zaproxy/zaproxy#1867]
This issue has been raised after discussing it here: https://github.com/zaproxy/zap-extensions/pull/30

The SOAP Scanner in alpha branch detects each operation correctly, but at the time of inserting them into the sites tree they are overwritten and only the last one prevails. 

This issue happens because all operations share the same URI, which is used to insert nodes into the sites tree and to cut repeated nodes.

I guess it could be fixed by adding some kind of optional ID to distinguish all operations sharing the same URI.

A workaround has been included in the plugin to make it work as long as the session remains active (it fails if you try to save the ZAP session and then restore it).

**Comment by psiinon** [comment]
The Sites tree is ZAP's understanding of the application.
Each node in the tree should represent a different 'page' or action/operation.
'Standard' web pages often handle multiple operations, and ZAP handles this by appending the parameter names to the node name, eg 
* GET page.html (param1, param2)
* GET page.html (param 3)
Maybe the SOAP scanner should include the operation name?
Possibly something like:
* mockTempConvertSoap12 (SOAP:FahrenheitToCelsius)

**Comment by Albertov91** [comment]
Hi @psiinon,

I have checked your suggestion about including the operation name as a parameter, but I have found some limitations.

The first problem is that SOAP messages are sent through POST method, so parameters must be included in the request message's body. Since request bodies start with ' <?xml version="1.0"... ' declaration, the plugin cannot add POST parameters because it would lead to malformed XML messages.

Is there any other way to reach the same solution, or am I missing something?
